optaplanner-wb-screens/optaplanner-wb-guided-rule-editor/optaplanner-wb-guided-rule-editor-client/src/main/java/org/optaplanner/workbench/screens/guidedrule/client/widget/ConstraintMatchInputWidgetBlurHandler.java

```diff
@@ -18,20 +18,75 @@
 
 import com.google.gwt.event.dom.client.BlurEvent;
 import com.google.gwt.event.dom.client.BlurHandler;
+import org.jboss.errai.ui.client.local.spi.TranslationService;
+
+import static org.optaplanner.workbench.screens.guidedrule.client.resources.i18n.GuidedRuleEditorConstants.ConstraintMatchInputWidgetBlurHandler_DoubleValueParsingError;
+import static org.optaplanner.workbench.screens.guidedrule.client.resources.i18n.GuidedRuleEditorConstants.ConstraintMatchInputWidgetBlurHandler_EmptyValuesAreNotAllowedForModifyScore;
+import static org.optaplanner.workbench.screens.guidedrule.client.resources.i18n.GuidedRuleEditorConstants.ConstraintMatchInputWidgetBlurHandler_IntegerValueParsingError;
+import static org.optaplanner.workbench.screens.guidedrule.client.resources.i18n.GuidedRuleEditorConstants.ConstraintMatchInputWidgetBlurHandler_LongValueParsingError;
 
 public class ConstraintMatchInputWidgetBlurHandler implements BlurHandler {
 
     private ConstraintMatchInputWidget widget;
+    private TranslationService translationService;
+    private String scoreHolderType;
 
-    public ConstraintMatchInputWidgetBlurHandler(ConstraintMatchInputWidget widget) {
+    public ConstraintMatchInputWidgetBlurHandler(final ConstraintMatchInputWidget widget,
+                                                 final TranslationService translationService,
+                                                 final String scoreHolderType) {
         this.widget = widget;
+        this.translationService = translationService;
+        this.scoreHolderType = scoreHolderType;
     }
 
     @Override
     public void onBlur(BlurEvent event) {
-        if (widget.getConstraintMatchValue() == null || widget.getConstraintMatchValue().trim().isEmpty()) {
-            widget.showEmptyValuesNotAllowedError();
+        String inputValue = widget.getConstraintMatchValue();
+        if (inputValue == null || inputValue.trim().isEmpty()) {
+            widget.showError(translationService.getTranslation(ConstraintMatchInputWidgetBlurHandler_EmptyValuesAreNotAllowedForModifyScore));
         } else {
+            inputValue = inputValue.trim();
+            if (inputValue.matches("-?\\s*\\d+(\\.\\d+)?.*")) {
```

Comment:
@mcimbora Hi.. I have nothing against using ```onBlur``` my main concern was about saving invalid content; however you seem OK with it so.... so am I :-) IIRC use of ```KeyUpHandler``` (or down or press) runs into other problems when people paste values into text boxes and you should really then think about how to handle floods of key presses too.